packages/aws-cdk/lib/api/util/sdk.ts

```diff
@@ -109,7 +118,11 @@ class CredentialsCache {
         const defaultAccount = await this.defaultAwsAccount.get();
         if (!awsAccountId || awsAccountId === defaultAccount) {
             debug(`Using default AWS SDK credentials for account ${awsAccountId}`);
-            return this.defaultCredentialProvider;
+
+            // CredentialProviderChain extends Credentials, but that is a lie.
+            // https://github.com/aws/aws-sdk-js/issues/2235
+            // Call resolve() instead.
+            return (await this.defaultCredentialProvider).resolvePromise();
```

Comment:
Are we going to break when this is fixed?

Reply:
They can't really fix it it in a way that will break us without breaking all of their other customers, which I imagine they won't do.

The fix is either "remove the extends" (my preference) or "implement the contract" (not my preference but least breakage-introducing). Both cases are fine for us.